apps/passport-client/components/screens/LoginScreens/LoginScreen.tsx

```diff
@@ -126,6 +130,93 @@ export function LoginScreen(): JSX.Element {
     [dispatch, email]
   );
 
+  const [tryStorageAccess, setTryStorageAccess] = useState(false);
+  const [tryStorageLogin, setTryStorageLogin] = useState(false);
+
+  /**
+   * Assuming we're in Chrome and an iframe, and we've successfully loaded an
+   * encryption key from local storage, try to use it to log in.
+   */
+  const tryToLogin = useCallback(
+    async (encryptionKey: string) => {
+      // Try to download and decrypt the storage
+      const storageRequest = await requestDownloadAndDecryptStorage(
+        appConfig.zupassServer,
+        encryptionKey
+      );
+      if (storageRequest.success) {
+        // Success, log in
+        dispatch({
+          type: "load-after-login",
+          storage: storageRequest.value,
+          encryptionKey
+        });
+      }
+    },
+    [dispatch]
+  );
+
+  /**
+   * This will only be called if we're in an iframe and Chrome.
+   */
+  const requestStorageAndLogIn = useCallback(async () => {
+    try {
+      // @ts-expect-error Chrome-only API
+      const handle: { localStorage: Storage } =
+        // @ts-expect-error Chrome-only API
+        await document.requestStorageAccess({ localStorage: true });
+
+      // Access granted, try reading the local storage
+      const encryptionKey = handle.localStorage.getItem("encryption_key");
+      if (encryptionKey) {
+        await tryToLogin(encryptionKey);
+      }
+    } catch (_e) {
+      // If the user rejected the storage access request, catch the exception
+      // but otherwise do nothing. The finally block will return the user to
+      // the regular login flow.
+    } finally {
+      setTryStorageAccess(false);
+      setTryStorageLogin(false);
+    }
+  }, [tryToLogin]);
+
+  useEffect(() => {
+    (async (): Promise<void> => {
+      // Are we in an iframe? If so, we might be able to skip requesting the
+      // user's email and password by retrieving their encryption key from the
+      // first-party local storage. Currently this only works on Chrome 125+.
+      const parser = new UAParser();
+      const browserName = parser.getBrowser().name;
+      const browserVersion = parser.getBrowser().version;
+      const isChrome125OrAbove =
```

Comment:
This should be factored out into a helper function (likely in a different file) like checkStorageAccesAPI.

Do you know how this behaves for things which are Chrome-based but not Chrome?  E.g. Microsoft Edge?  Or embedded web apps based on Chromium such as Electron?  If those other situations have the storage access API, and we can properly detect them here, it broadens the set of users/cases we can handle.  The docs I can find suggest that some non-Chrome browsers like Opera already support this: https://developer.mozilla.org/en-US/docs/Web/API/Document/requestStorageAccess

It would be good if we could directly detect the presence of the feature rather than trying to figure out the browser version, but I don't see an obvious way to do that.  We can't just whether `document.hasStorageAccess` and `document.requestStorageAccess` are defined, because they always are, and what we need to know is whether the types.localStorage field is supported.

Reply:
I've tested UAParser with other Chrome-based browsers and it generally returns different names for browsers which are Chrome-based not actually Chrome (e.g. Edge, Brave). Interestingly Arc is identified as Chrome, but this seems to be because it really is just Chrome with a new UI, and has an identical feature-set to Chrome given equivalent version numbers. The `requestStorageAccess` function works identically in Arc as it does in Chrome.

Reply:
Oh, and I see in the docs that Edge doesn't actually support local storage access, so I guess we don't want to know if it's "Chrome based".
A potential solution here is "just ask and see what we get", but that has the downside that the user sees a request.
That approach might work out if we were requesting cookie access anyway for some useful purpose (like skipping email verification as we discussed elsewhere), in which case doing the local storage access request could be an optional part of the same user interaction.  But that's an option for later, not this PR.